api/v1/configurationpolicy_types.go

```diff
@@ -84,29 +85,34 @@ func (t Target) String() string {
 }
 
 // EvaluationInterval configures the minimum elapsed time before a configuration policy is
-// reevaluated. If the policy spec is changed, or if the list of namespaces selected by the policy
+// reevaluated. This defaults to "watch" to leverage Kubernetes API watches instead of polling the Kubernetes API
```

Comment:
I have a question. When it is watch, how policy detect and change the status when the resource which is created by configuration policy is changed, created or deleted (ex: change status from compliant -> noncompliant)

Reply:
Any query using the `DynamicWatcher` client or from a template function executed in `TemplateResolver` causes a watch for that `ConfigurationPolicy` and triggers a reconcile of the `ConfigurationPolicy` when any of those objects change. Those reconciles are triggered from the additional source added in the `SetupWithManager` method.